packages/lib/src/__tests__/basic-io.test.ts

```diff
@@ -0,0 +1,189 @@
+import { CLIError } from '@oclif/errors'
+
+import { inputAndOutputItem, Naming, outputItem, outputList, Sorting } from '../basic-io'
+import * as format from '../format'
+import { InputProcessor, UserInputProcessor } from '../input'
+import * as inputBuilder from '../input-builder'
+import { IOFormat } from '../io-util'
+import * as output from '../output'
+import * as outputBuilder from '../output-builder'
+import { SmartThingsCommandInterface } from '../smartthings-command'
+import { buildMockCommand } from './test-lib/mock-command'
+import { SimpleType } from './test-lib/simple-type'
+
+
+describe('basic-io', () => {
+	const item = { 'str': 'string', num: 5 }
+	const list = [item]
+	let baseCommand: SmartThingsCommandInterface
+	let formatAndWriteItemSpy: jest.SpyInstance<Promise<void>, [command: (SmartThingsCommandInterface & format.TableCommonOutputProducer<unknown>) | (SmartThingsCommandInterface & format.CustomCommonOutputProducer<unknown>), item: unknown, defaultIOFormat?: IOFormat | undefined]>
+	let formatAndWriteListSpy: jest.SpyInstance<Promise<void>, [command: (SmartThingsCommandInterface & Sorting & Naming) | (SmartThingsCommandInterface & format.TableCommonListOutputProducer<unknown> & Naming) | (SmartThingsCommandInterface & format.CustomCommonListOutputProducer<unknown> & Naming), list: unknown[], includeIndex?: boolean | undefined]>
+
+	beforeEach(() => {
+		baseCommand = {
+			...buildMockCommand(),
+			flags: {
+				output: 'output.yaml',
+			},
+		}
+		formatAndWriteItemSpy = jest.spyOn(format, 'formatAndWriteItem')
+		formatAndWriteItemSpy.mockImplementation(async () => { /* empty */})
+
+		formatAndWriteListSpy = jest.spyOn(format, 'formatAndWriteList')
+		formatAndWriteListSpy.mockImplementation(async () => { /* empty */})
+	})
+
+	afterEach(() => {
+		jest.clearAllMocks()
+	})
+
+	describe('outputItem', () => {
+		it('gets data and calls formatAndWriteItem', async () => {
+			const command = {
+				...baseCommand,
+				tableFieldDefinitions: [],
+			}
+
+			const getDataMock = jest.fn()
+			getDataMock.mockResolvedValue(item)
+
+			const result = await outputItem<SimpleType>(command, getDataMock)
+
+			expect(result).toBe(item)
+			expect(formatAndWriteItemSpy).toHaveBeenCalledTimes(1)
+			expect(formatAndWriteItemSpy).toHaveBeenLastCalledWith(command, item)
+		})
+	})
+
+	describe('outputList', () => {
+		it('gets data and calls formatAndWriteList', async () => {
+			const command = {
+				...baseCommand,
+				listTableFieldDefinitions: [],
+				primaryKeyName: 'num',
+				sortKeyName: 'str',
+
+			}
+
+			const getDataMock = jest.fn()
+			getDataMock.mockResolvedValue(list)
```

Comment:
These can be achieved in line as well
```typescript
const getDataMock = jest.fn().mockResolvedValue(list)
```

Reply:
I've updated to make use of this now in several places, including my spies.